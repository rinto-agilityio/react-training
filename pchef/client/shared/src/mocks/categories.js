export const categories = [
  {
    id: 'chicken',
    name: 'Chicken',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Heathy',
    name: 'Heathy',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Easy Breakast',
    name: 'Easy Breakast',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Dessert',
    name: 'Dessert',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Fish',
    name: 'Fish',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Pizza',
    name: 'Pizza',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Cake',
    name: 'Cake',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Deserts',
    name: 'Deserts',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Salad',
    name: 'Salad',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Kids',
    name: 'Kids',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Meat',
    name: 'Meat',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Easy Lunch',
    name: null,
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
]
