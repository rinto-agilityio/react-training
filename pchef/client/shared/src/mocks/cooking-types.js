export const cookingTypes = [
  {
    id: 'baking',
    name: 'Baking',
  },
  {
    id: 'roasting',
    name: 'Roasting',
  },
  {
    id: 'frying',
    name: 'Frying',
  },
]
