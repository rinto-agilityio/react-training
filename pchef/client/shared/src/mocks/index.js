// mock categories
export const category = {
  name: 'Heathy',
  imgUrl:
    'https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500',
}
