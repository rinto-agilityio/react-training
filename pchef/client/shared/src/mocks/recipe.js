export const recipe = {
  description: '2lb pork neck bones, 2lb beef bones, 1lb oxtail',
  title: 'Bun Bo Hue Recipe',
  imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
  votes: [1],
  steps: [
    {
      step: 1,
      description: 'Step 1 description',
      title: 'Step 1 title',
      subTitle: 'Step 1 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 2,
      description: 'Step 2 description',
      title: 'Step 2 title',
      subTitle: 'Step 2 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 3,
      description: 'Step 3 description',
      title: 'Step 3 title',
      subTitle: 'Step 3 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 4,
      description: 'Step 4 description',
      title: 'Step 4 title',
      subTitle: 'Step 4 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 5,
      description: 'Step 5 description',
      title: 'Step 5 title',
      subTitle: 'Step 5 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 6,
      description: 'Step 6 description',
      title: 'Step 6 title',
      subTitle: 'Step 6 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
  ],
}
