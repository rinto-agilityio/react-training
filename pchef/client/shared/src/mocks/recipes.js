export const recipes = [
  {
    id: 0,
    title: 'Bun Bo Hue Recipe',
    subTitle: 'Traditional food',
    description: '2lb pork neck bones, 2lb beef bones, 1lb oxtail',
    imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    votes: [1],
    views: 1,
    steps: [
      {
        step: 1,
        description: 'Step 1 description',
        title: 'Step 1 title',
        subTitle: 'Step 1 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 2,
        description: 'Step 2 description',
        title: 'Step 2 title',
        subTitle: 'Step 2 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 3,
        description: 'Step 3 description',
        title: 'Step 3 title',
        subTitle: 'Step 3 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 4,
        description: 'Step 4 description',
        title: 'Step 4 title',
        subTitle: 'Step 4 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 5,
        description: 'Step 5 description',
        title: 'Step 5 title',
        subTitle: 'Step 5 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 6,
        description: 'Step 6 description',
        title: 'Step 6 title',
        subTitle: 'Step 6 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
    ],
  },
  {
    id: 1,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
  {
    id: 2,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
  {
    id: 3,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
]
