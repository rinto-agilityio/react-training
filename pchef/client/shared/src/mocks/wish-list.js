export const wishList = [{
  id: '1',
  date: Date.now(),
  categoryId: '1',
  cookingTypeId: '1',
},
{
  id: '2',
  date: Date.now(),
  categoryId: '2',
  cookingTypeId: '2',
}]
