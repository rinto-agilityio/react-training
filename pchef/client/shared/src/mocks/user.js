import { categories } from './categories'
import { recipes } from './recipes'

const favoriteRecipeId: Array<string> = recipes.map(item => item.id)
export const user = {
  id: '1',
  name: 'Alex',
  email: 'alex@gmail.com',
  avatar:
    'https://cdn4.iconfinder.com/data/icons/avatars-21/512/avatar-circle-human-female-black-7-512.png',
  favoriteRecipe: favoriteRecipeId,
  followCategory: categories,
}
