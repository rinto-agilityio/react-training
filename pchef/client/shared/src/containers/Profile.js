import { compose } from 'react-apollo'

// GraphQL
import { getUser } from '../graphql/user'
import { userToggleRecipe, userToggleVote } from '../graphql/user'
import Profile from '../screens/profile'

export default compose(getUser, userToggleRecipe, userToggleVote)(Profile)
