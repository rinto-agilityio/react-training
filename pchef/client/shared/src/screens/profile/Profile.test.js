// Mock
import { user, recipes, categories } from '../../mocks'

// Components
import Profile from '.'

const props = {
  data: {
    user,
    followCategory: categories,
    favoriteRecipe: recipes,
  },
  loading: false,
  error: undefined,
}

describe('Screen', () => {
  describe('Profile', () => {
    let component

    beforeEach(() => {
      component = shallow(<Profile {...props} />)
    })

    it('should render Category component with grid', () => {
      const renderComponent = renderer.create(<Profile {...props} />)
      expect(renderComponent).toMatchSnapshot()
    })

    it('should render Loading component when query is loading', () => {
      component.setProps({ loading: true })
      expect(component.find('Loading')).toHaveLength(1)
    })

    it('should render Error when query failed', () => {
      component.setProps({ error: 'Failed' })
      expect(component.find('Error')).toHaveLength(1)
    })
  })
})
