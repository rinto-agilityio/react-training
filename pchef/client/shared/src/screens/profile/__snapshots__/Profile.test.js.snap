// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Profile should render Category component with grid 1`] = `
<div
  className="css-view-1dbjc4n r-backgroundColor-14lw9ot r-width-13qz1uu"
>
  <div
    className="css-view-1dbjc4n r-backgroundColor-kemksi r-opacity-1g80hic"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1habvwh r-height-7r4507 r-justifyContent-17s6mgv r-marginLeft-1jj8364 r-marginRight-lchren r-maxWidth-c3zab1 r-paddingHorizontal-d9fdf6 r-width-13qz1uu"
    >
      <div
        className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        tabIndex="0"
      >
        <div
          className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
          style={
            Object {
              "borderBottomColor": "rgba(255,255,255,1.00)",
              "borderBottomLeftRadius": "50%",
              "borderBottomRightRadius": "50%",
              "borderBottomWidth": "3px",
              "borderLeftColor": "rgba(255,255,255,1.00)",
              "borderLeftWidth": "3px",
              "borderRightColor": "rgba(255,255,255,1.00)",
              "borderRightWidth": "3px",
              "borderTopColor": "rgba(255,255,255,1.00)",
              "borderTopLeftRadius": "50%",
              "borderTopRightRadius": "50%",
              "borderTopWidth": "3px",
              "height": "80px",
              "marginBottom": "-10px",
              "width": "80px",
            }
          }
        >
          <div
            className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
            style={
              Object {
                "backgroundImage": "url(\\"https://cdn4.iconfinder.com/data/icons/avatars-21/512/avatar-circle-human-female-black-7-512.png\\")",
              }
            }
          />
          <img
            alt=""
            className="css-accessibilityImage-9pa8cd"
            draggable={false}
            src="https://cdn4.iconfinder.com/data/icons/avatars-21/512/avatar-circle-human-female-black-7-512.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-backgroundColor-14lw9ot r-flexDirection-18u37iz r-justifyContent-1wtj0ep r-marginLeft-1jj8364 r-marginRight-lchren r-maxWidth-c3zab1 r-padding-1pcd2l5 r-width-13qz1uu r-zIndex-1wyyakw"
  >
    <div
      className="css-text-901oao r-fontSize-adyw6z r-fontWeight-1kfrs79"
      dir="auto"
    >
      Alex
    </div>
    <div
      className="css-view-1dbjc4n r-backgroundColor-14lw9ot"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxPack": "center",
          "WebkitJustifyContent": "center",
          "alignItems": "center",
          "borderBottomColor": "rgba(0,0,0,1.00)",
          "borderBottomLeftRadius": "20px",
          "borderBottomRightRadius": "20px",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "1px",
          "borderLeftColor": "rgba(0,0,0,1.00)",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "1px",
          "borderRightColor": "rgba(0,0,0,1.00)",
          "borderRightStyle": "solid",
          "borderRightWidth": "1px",
          "borderTopColor": "rgba(0,0,0,1.00)",
          "borderTopLeftRadius": "20px",
          "borderTopRightRadius": "20px",
          "borderTopStyle": "solid",
          "borderTopWidth": "1px",
          "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
          "justifyContent": "center",
          "minWidth": "64px",
          "msFlexAlign": "center",
          "msFlexPack": "center",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
        }
      }
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-overflow-1udh08x r-touchAction-1otgn73 r-userSelect-lrvibr"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        style={
          Object {
            "borderBottomLeftRadius": "4px",
            "borderBottomRightRadius": "4px",
            "borderTopLeftRadius": "4px",
            "borderTopRightRadius": "4px",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1777fci"
        >
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "direction": "ltr",
                "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              }
            }
          >
            <span
              className="css-text-901oao css-textHasAncestor-16my406 r-color-cqee49 r-fontWeight-1kfrs79 r-textTransform-3twk1y"
              dir="auto"
              style={
                Object {
                  "direction": "ltr",
                  "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "textAlign": "left",
                }
              }
            >
              Edit profile
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-marginLeft-1jj8364 r-marginRight-lchren r-width-13qz1uu"
  >
    <div
      className="css-view-1dbjc4n r-backgroundColor-14lw9ot r-flexDirection-18u37iz r-marginLeft-1jj8364 r-marginRight-lchren r-maxWidth-c3zab1 r-padding-1pcd2l5 r-paddingBottom-1mdbw0j r-width-13qz1uu"
    >
      <div
        className="css-view-1dbjc4n r-borderBottomWidth-qklmqi r-borderColor-1xc7w19 r-borderStyle-1phboty r-cursor-1loqt21 r-paddingBottom-1inuy60 r-paddingLeft-1hfyk0a r-touchAction-1otgn73 r-width-1iadsmv"
        data-focusable={true}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        tabIndex="0"
      >
        <div
          className="css-text-901oao r-fontWeight-majxgm"
          dir="auto"
        >
          0
        </div>
        <div
          className="css-text-901oao r-fontWeight-majxgm"
          dir="auto"
        >
          recipes
        </div>
      </div>
      <div
        className="css-view-1dbjc4n r-borderBottomWidth-qklmqi r-borderColor-42olwf r-borderStyle-1phboty r-cursor-1loqt21 r-paddingBottom-1inuy60 r-paddingLeft-1hfyk0a r-touchAction-1otgn73 r-width-1iadsmv"
        data-focusable={true}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        tabIndex="0"
      >
        <div
          className="css-text-901oao r-fontWeight-majxgm"
          dir="auto"
        >
          4
        </div>
        <div
          className="css-text-901oao r-fontWeight-majxgm"
          dir="auto"
        >
          favorites
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-backgroundColor-n2h5ot r-flexDirection-18u37iz r-flexWrap-1w6e6rj r-marginLeft-1jj8364 r-marginRight-lchren r-maxWidth-c3zab1 r-minHeight-fo66li r-padding-1pcd2l5 r-width-13qz1uu"
    >
      <div
        className="css-text-901oao"
        dir="auto"
      >
        No recipes to show
      </div>
    </div>
  </div>
</div>
`;
